### PR TITLE
Added Google Cloud Spanner metrics receiver skeleton.

### PR DESCRIPTION
**Pull request by ydrozhdzhal**
Added Google Cloud Spanner metrics receiver skeleton.

### CODE REVIEW

**Comment by bogdandrutu** [comment]
Hey @ydrozhdzhal please sign the CLA

**Comment by ydrozhdzhal** [comment]
> Hey @ydrozhdzhal please sign the CLA
@bogdandrutu , I am in progress with this. I hope to manage this tomorrow.

**Comment by ydrozhdzhal** [comment]
> Hey @ydrozhdzhal please sign the CLA

Done

**Comment by ydrozhdzhal** [comment]
@bogdandrutu , we signed CLA

File: receiver/googlecloudspannerreceiver/config.go
Comment (anuraaga) [inline receiver/googlecloudspannerreceiver/config.go:27]: How about using a struct `TopMetricsQuery` with a field inside `max_rows` to allow more options to be added later

File: receiver/googlecloudspannerreceiver/config.go
Comment (anuraaga) [inline receiver/googlecloudspannerreceiver/config.go:28]: Is it idiomatic for one collector to be scraping multiple projects like this? I figured it's more idiomatic if each project would have a collector, that may export their metrics in OTLP to a collector in a monitoring account if cross-project is needed.

This is fine if it seems conventional for GCP though

File: receiver/googlecloudspannerreceiver/config.go
Comment (anuraaga) [inline receiver/googlecloudspannerreceiver/config.go:33]: Not supporting multiple projects would allow using google default credentials I think which could be more user-friendly

File: receiver/googlecloudspannerreceiver/config.go
Comment (anuraaga) [inline receiver/googlecloudspannerreceiver/config.go:70]: Make the sub-validation helpers private instead of public (`validate`)

File: receiver/googlecloudspannerreceiver/config.go
Comment (ydrozhdzhal) [reply receiver/googlecloudspannerreceiver/config.go:27]: Nice idea. For now we do not expect more options.
We can introduce such structure later if required(more options appear).

File: receiver/googlecloudspannerreceiver/config.go
Comment (ydrozhdzhal) [reply receiver/googlecloudspannerreceiver/config.go:28]: Yes, I agree with you, but:

- current configuration allows to reach the same result
- our use cases require to have one collector for multiple projects with single receiver.

File: receiver/googlecloudspannerreceiver/config.go
Comment (ydrozhdzhal) [reply receiver/googlecloudspannerreceiver/config.go:33]: I agree but in our use case it is required to have multiple projects.

File: receiver/googlecloudspannerreceiver/config.go
Comment (ydrozhdzhal) [reply receiver/googlecloudspannerreceiver/config.go:70]: Ok, will do

Comment (ydrozhdzhal) [reply receiver/googlecloudspannerreceiver/config.go]: Done.

**Comment by ydrozhdzhal** [comment]
@anuraaga , I replied to you comments and made changes/updates.

**Comment by ydrozhdzhal** [comment]
@balachandr, you asked to add you here also.

**Review by anuraaga — APPROVED** [review]

**Comment by ydrozhdzhal** [comment]
@bogdandrutu what are the next steps? Who can approve workflow for this MR? Do I need to squash(not squash) commits and rebase?

**Review by bogdandrutu — APPROVED** [review]

**Comment by ydrozhdzhal** [comment]
@bogdandrutu there were failing tests not related to my code updates, so I rebased branch to the latest main and squashed some commits. Can you approve workflow one more time? Will it be merged automatically or more actions are required?

**Comment by ydrozhdzhal** [comment]
@bogdandrutu there are failing tests not related to my code updates, so I rebased branch to the latest main and squashed some commits but still no luck.

**Comment by ydrozhdzhal** [comment]
@bogdandrutu , @anuraaga looks like workflow needs to be approved again, after re-basing no failures in workflow at last.

File: receiver/googlecloudspannerreceiver/config.go
Comment (punya) [reply receiver/googlecloudspannerreceiver/config.go:34]: For this and https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/5240/files#r709960450, could you describe the use cases in some more detail?

**Comment by ydrozhdzhal** [comment]
@bogdandrutu Who can press the merge button? Also I am not sure about this one failing test in workflow - looks like it is unrelated to my changes.

File: receiver/googlecloudspannerreceiver/config.go
Comment (ydrozhdzhal) [reply receiver/googlecloudspannerreceiver/config.go:34]: @punya , current configuration options allows you either work with one or multiple projects. Also there was request to us that potential users do not want to have multiple receivers and multiple pipelines configured in scope of one collector instance for Spanner monitoring. They want to have possibility to collect data from all their projects in scope of one receiver instance.

**Pull request merged**
